hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HiveTypeUtils.java

```diff
@@ -181,13 +182,13 @@ public static TypeInfo generateTypeInfo(Schema schema,
       return TypeInfoFactory.getVarcharTypeInfo(maxLength);
     }
 
-    if (type == INT
-        && AvroSerDe.DATE_TYPE_NAME.equals(schema.getProp(AvroSerDe.AVRO_PROP_LOGICAL_TYPE))) {
+    if (type == DATE
+        && AvroSerDe.DATE_TYPE_NAME.equals((String) schema.getProp(AvroSerDe.AVRO_PROP_LOGICAL_TYPE))) {
       return TypeInfoFactory.dateTypeInfo;
     }
 
     if (type == LONG
-        && AvroSerDe.TIMESTAMP_TYPE_NAME.equals(schema.getProp(AvroSerDe.AVRO_PROP_LOGICAL_TYPE))) {
```

Comment:
Should this check `TIMESTAMP` type instead of `LONG`?

Reply:
Yeap, good catch, let me modify this.